File: bench/src/fibonacci/fibonacci.py
#!/usr/bin/env python3
"""Fibonacci iterative - Python
Reads BENCH_INPUT env var for input size, defaults to 1000"""

import os

def fibonacci(n: int) -> int:
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

def main():
    n = int(os.environ.get("BENCH_INPUT", "1000"))
    result = 0
    for _ in range(10000):
        result = fibonacci(n)
    print(result)

if __name__ == "__main__":
    main()

File: bench/src/fibonacci/fibonacci.rs
//! Fibonacci iterative - Rust
//! Reads BENCH_INPUT env var for input size, defaults to 1000

use std::env;

fn fibonacci(n: u64) -> u64 {
    if n <= 1 {
        return n;
    }
    let mut a = 0;
    let mut b = 1;
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    let n: u64 = env::var("BENCH_INPUT")
        .ok()
        .and_then(|s| s.parse().ok())
        .unwrap_or(1000);

    let mut result = 0;
    for _ in 0..10000 {
        result = fibonacci(n);
    }
    println!("{}", result);
}